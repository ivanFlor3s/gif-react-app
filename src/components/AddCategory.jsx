import React, { useState } from "react";
import PropTypes from 'prop-types'


export const AddCategory = ({addGifHandler}) => {
  const [inputValue, setInputValue] = useState("");

  const changeHandler = (e) => {
    setInputValue(e.target.value);
  };

  const handleSubmit = (e) => {
      e.preventDefault()

      addGifHandler((categorias = []) => {
          if(inputValue.trim().length < 3){
              return [...categorias]
          }
          else if(!categorias.includes(inputValue)){
            setInputValue('')
            return [...categorias,inputValue]
          }
          else{
              console.error('Ya se agrego la categoria', inputValue)
              return [...categorias]
          }
        })
  }

  return (
    <form onSubmit={handleSubmit}>
      <label>Nueva categoria </label>
      <input
        className="inputCategoria"
        type="text"
        value={inputValue}
        onChange={changeHandler}
        placeholder="Ingrese una categoria"
      />
    </form>
  );
};

AddCategory.propTypes = {
    addGifHandler: PropTypes.func.isRequired
}