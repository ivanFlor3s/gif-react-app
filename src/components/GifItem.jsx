import React from 'react'
import PropTypes from 'prop-types'

export const GifItem = ({img, title}) => {
    return (
        <div className="card animate__animated  animate__fadeInLeftBig">
          <div>
              <img  className="card__img" src={img} alt={title} />
          </div>
          <div className="card__footer">
              <h4>{title}</h4>
          </div>
      </div>
    )
} 

GifItem.proptype = {
    img: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
}
