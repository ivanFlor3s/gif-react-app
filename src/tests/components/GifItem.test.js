import React from "react";
import { shallow } from "enzyme";
import { GifItem } from "../../components/GifItem";

describe('Pruebas en GifItem component', () => {

    test('Debe mostrar el componente correctamente', () => {
        const [title, url] = ['Mi titulo', 'https://esto+es+una+url.jpg']
        const wrapper = shallow(<GifItem title={title} img={url}></GifItem>)
        expect(wrapper).toMatchSnapshot()
    })
})