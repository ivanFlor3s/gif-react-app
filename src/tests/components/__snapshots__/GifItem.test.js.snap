// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifItem component Debe mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated  animate__fadeInLeftBig"
>
  <div>
    <img
      alt="Mi titulo"
      className="card__img"
      src="https://esto+es+una+url.jpg"
    />
  </div>
  <div
    className="card__footer"
  >
    <h4>
      Mi titulo
    </h4>
  </div>
</div>
`;
